share_app/src/main/java/org/odk/share/controller/WifiHelper.java

```diff
@@ -11,11 +18,77 @@
 public class WifiHelper {
 
     private final WifiManager wifiManager;
+    public static WifiHelper wifiHelper;
+    private Context context;
 
-    public WifiHelper(Context context) {
+    private WifiHelper(Context context) {
+        this.context = context;
         wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
     }
 
+    public static synchronized WifiHelper getInstance(Context context) {
```

Comment:
Creating a singleton is not a good practice and later leads to complications. Is there any specific use case because of which you had to do it? If not then please try not to use it